shared/actions/profile.js

```diff
@@ -624,25 +605,27 @@ function * dropPgpSaga (action: DropPgp): SagaGenerator<any, any> {
 function * generatePgpSaga (): SagaGenerator<any, any> {
   yield put(routeAppend('generate'))
 
+  const channelConfig = singleFixedChannelConfig(['keybase.1.pgpUi.keyGenerated', 'keybase.1.pgpUi.shouldPushPrivate', 'keybase.1.pgpUi.finished', 'finished'])
+
   // $ForceType
   const pgpInfo: PgpInfo = yield select(({profile: {pgpInfo}}: TypedState) => pgpInfo)
-  const generatePgpKeyChan = yield call(generatePgpKey, pgpInfo)
+  // $ForceType
+  const generatePgpKeyChanMap: ChannelMap<any> = yield call(generatePgpKey, channelConfig, pgpInfo)
 
   try {
     // $ForceType
-    const {cancel, keyGenerated}: {keyGenerated: IncomingKeyGenerated, cancel: ?any} = yield race({
-      keyGenerated: take(generatePgpKeyChan, 'keybase.1.pgpUi.keyGenerated'),
+    const {cancel, keyGenerated}: {keyGenerated: any, cancel: ?any} = yield race({
+      keyGenerated: takeFromChannelMap(generatePgpKeyChanMap, 'keybase.1.pgpUi.keyGenerated'),
       cancel: take(Constants.cancelPgpGen),
     })
 
     if (cancel) {
-      generatePgpKeyChan && generatePgpKeyChan.close()
+      closeChannelMap(generatePgpKeyChanMap)
```

Comment:
This probably needs to exit the outer flow, since the lower lines expect a valid channel map.